pocs/utils/rs232.py

```diff
@@ -33,38 +36,41 @@ def __init__(self,
         """
         PanBase.__init__(self)
 
-        try:
-            self.ser = serial.Serial()
-            self.ser.port = port
-            self.ser.baudrate = baudrate
-
-            self.ser.bytesize = serial.EIGHTBITS
-            self.ser.parity = serial.PARITY_NONE
-            self.ser.stopbits = serial.STOPBITS_ONE
-            self.ser.timeout = 1.0
-            self.ser.xonxoff = False
-            self.ser.rtscts = False
-            self.ser.dsrdtr = False
-            self.ser.write_timeout = False
-            self.ser.open()
-
-            self.name = name
-
-            self.logger.debug(
-                'Serial connection set up to {}, sleeping for two seconds'.format(
-                    self.name))
-            time.sleep(2)
-            self.logger.debug('SerialData created')
-        except Exception as err:
-            self.ser = None
-            self.logger.critical('Could not set up serial port {} {}'.format(
-                port, err))
+        if not port:
+            raise ValueError('Must specify port for SerialData')
+
+        self.name = name
+        self.retry_limit = retry_limit
+        self.retry_delay = retry_delay
+
+        self.ser = serial.serial_for_url(port, do_not_open=True)
+
+        # Configure the PySerial class.
+        self.ser.baudrate = baudrate
+        self.ser.bytesize = serial.EIGHTBITS
+        self.ser.parity = serial.PARITY_NONE
+        self.ser.stopbits = serial.STOPBITS_ONE
+        self.ser.timeout = 1.0
+        self.ser.xonxoff = False
+        self.ser.rtscts = False
+        self.ser.dsrdtr = False
+        self.ser.write_timeout = False
+
+        # Properties have been set to reasonable values, ready to open the port.
+        self.ser.open()
+
+        # TODO(jamessynge): Consider eliminating this sleep period, or making
```

Comment:
I still need to dig back through my notes about why the delay  is there in the first place but would prefer to capture these TODOs in an Issue unless it will be resolved as part of this PR>

Reply:
Ah. Sorry, I'm in the habit of capturing these things in the code where their context is readily apparent.

Reply:
Issue added, TODO removed.